JDBC/OAuthSampleApp/src/main/java/OAuthSampleApp.java

```diff
@@ -10,91 +10,329 @@
+		String CONNECTION_STRING = prop.getProperty("ConnectionString");    
+		String USER = prop.getProperty("User");
+		String CLIENT_ID = prop.getProperty("ClientId");
+		String CLIENT_SECRET = prop.getProperty("ClientSecret");
+		String INTROSPECT_URL = prop.getProperty("TokenUrl") + "introspect";
+		Statement st = vConnection.createStatement();
+		st.executeUpdate("DROP USER IF EXISTS " + USER + " CASCADE;");
+		st.executeUpdate("DROP AUTHENTICATION IF EXISTS v_oauth CASCADE;");
+		st.executeUpdate("CREATE AUTHENTICATION v_oauth METHOD 'oauth' HOST '0.0.0.0/0';");
+		st.executeUpdate("ALTER AUTHENTICATION v_oauth SET client_id= '" + CLIENT_ID + "';");
+		st.executeUpdate("ALTER AUTHENTICATION v_oauth SET client_secret= '" + CLIENT_SECRET + "';");
+		st.executeUpdate("ALTER AUTHENTICATION v_oauth SET introspect_url = '" + INTROSPECT_URL + "';");
+		st.executeUpdate("CREATE USER " + USER + ";");
+		st.executeUpdate("GRANT AUTHENTICATION v_oauth TO " + USER + ";");
+		st.executeUpdate("GRANT ALL ON SCHEMA PUBLIC TO " + USER +";");
+		st.close();
+	}
+	// Dispose the authentication record from database
+	private static void TearDown() 
+	{
+		try
+		{
+			Statement st = vConnection.createStatement(); 
+			String USER = prop.getProperty("User");
+			st.executeUpdate("DROP USER IF EXISTS " + USER + " CASCADE");
+			st.executeUpdate("DROP AUTHENTICATION IF EXISTS v_oauth CASCADE");
+			vConnection.close();
+		}catch(Exception e)
+		{
+			e.printStackTrace();
+		}
+	}
+	// Connect to Database using access Token
+	private static Connection connectToDB(String accessToken) throws SQLException 
+	{
+		Properties jdbcOptions = new Properties();
+		jdbcOptions.put("oauthaccesstoken", accessToken);
+		return DriverManager.getConnection(
+				"jdbc:vertica://" + getParam("Host") + ":" + getParam("Port") + "/" + getParam("Database") , jdbcOptions);
     }
-
-    private static ResultSet executeQuery(Connection conn) throws SQLException {
-        Statement stmt = conn.createStatement();
-        return stmt.executeQuery("SELECT user_id, user_name FROM USERS ORDER BY user_id");
+	// Test connection using access token and test database query result
+	private static void ConnectToDatabase() throws SQLException 
+	{
+		int connAttemptCount = 0;
+		while (connAttemptCount <= 1)
+        {
+            try
+            {
+            	String accessToken = System.getProperty(OAUTH_ACCESS_TOKEN_VAR_STRING);
+        		if( null == accessToken || accessToken.isEmpty())
+        		{
+					throw new Exception("Access Token is not available.");
+        		}else
+        		{
+        			Connection conn = connectToDB(accessToken);
+        			ResultSet rs = executeQuery(conn);
+        			printResults(rs);
+					break;
+        		}
+            }catch(Exception ex) 
+            {
+            	if (connAttemptCount > 0) 
+            	{ 
+            		break;
+            	}
+            	try {
+					ex.printStackTrace();
```

Comment:
I don't think we should be printing the stack trace here or on line 148. At this point in the flow, we are just refreshing the tokens, nothing 'wrong' has heppened.

Reply:
During compilation, It's giving an error for this code that it has an unhandled exception at this line. Due to this it needs to handle the exception and need to avoid the use of the "System.out.println" (console output) in the code as per discussion with Sandeep. Because of that whenever we need error details, I have used printStackTrace.

Reply:
When an access token is expired and you get a refresh token, is the stack trace printed out? That would be confusing to the user and what we want to avoid.

Reply:
Removed printing of stack traces while getting tokens in connect to database.